collections/src/main/scala/strawman/collection/immutable/WrappedString.scala

```diff
@@ -0,0 +1,63 @@
+package strawman.collection
+package immutable
+
+import scala.{Char, Int}
+import scala.Predef.String
+import mutable.{Builder, StringBuilder}
+
+/**
+  *  This class serves as a wrapper augmenting `String`s with all the operations
+  *  found in indexed sequences.
+  *
+  *  The difference between this class and `StringOps` is that calling transformer
+  *  methods such as `filter` and `map` will yield an object of type `WrappedString`
+  *  rather than a `String`.
+  *
+  *  @param self    a string contained within this wrapped string
+  *
+  *  @since 2.8
+  *  @define Coll `WrappedString`
+  *  @define coll wrapped string
+  */
+final class WrappedString(val self: String) extends AbstractSeq[Char] with IndexedSeq[Char] {
```

Comment:
Maybe it’s time to have `AbstractIndexedSeq`?

Reply:
This is the same as in the old collections. There's `collection.AbstractSeq` but no `collection.AbstractIndexedSeq` or `immutable.Abstract*`. I'll do some more experiments and possibly open another PR. First tests show that `immutable.AbstractSeq` could be worth it, but `immutable.AbstractIndexedSeq` is probably not.